conans/model/editable_cpp_info.py

```diff
@@ -44,22 +57,30 @@ def load(filepath):
         except configparser.Error as e:
             raise ConanException("Error parsing layout file: %s\n%s" % (filepath, str(e)))
         data = OrderedDict()
+        folders = {}
```

Comment:
We don't need a new member to store these _folders_, they can go just into the _data_ one.

Reply:
No, better not mix it:

```python
if data:  # Invalidate previously existing dirs
            for info_dir in self.cpp_info_dirs:
                setattr(cpp_info, info_dir, [])
```
The definition of a ``build_folder`` would invalidate includedirs, etc, and I think this is not desired.

Reply:
I would double check this. This is against the principle we established when talking about the editable layouts in the whiteboard: only one source of truth. IMO it is ok to invalidate the `cpp_info dirs` also when it is only the `build_folder` (or the source one) in the layout file.

Reply:
Not nice, because you cannot define those folders in the ``package_info()`` and ``cpp_info``. So you can have a perfectly valid, customized ``package_info()``, but you still need to define your ``build_folder`` in the layout, and certainly you don't want to invalidate the ``cpp_info``. 
One thing is layout (build_folder, source_folder), and the other is cpp_info.

Reply:
I understand your concern, but still I think that two sources of truth can be confusing. With the _double check_ I wanted to share this consideration with the team, to see what they think about it. 

Please, @conan-io/barbarians , your opinion on this thread?

Reply:
It is not that it would be using 2 sources of truth. Not allowing this will forbid the possibility to use the information from ``package_info()`` and ``cpp_info``, as the ``build_folder`` is necessary for workspaces, and that cannot be defined in ``package_info()``.

Reply:
I agree @memsharded. 
Shouldn't this module be renamed too? Do you know if doing "git rename" the diff is smart enough?

Reply:
Yeah, I didn't want to rename it yet (same with WorkspaceTest), because when there are % of changes, then it detects as a new file, and it is impossible to review. I will do those renames and moves after, in another PR.